docs: add explanation to .env (#3993)
web-infra-dev/rsbuild: website/docs/en/guide/advanced/env-vars.mdx

## `.env` File

> When a `.env` file exists in the project root directory, Rsbuild CLI will automatically use [dotenv](https://npmjs.com/package/dotenv) to load these env variables and add them to the current Node.js process, The [Public Variables](#public-variables) will be exposed in the client code.

Reviewer: ```suggestion
When a `.env` file exists in the project root directory, Rsbuild CLI will automatically use [dotenv](https://npmjs.com/package/dotenv) to load these env variables and add them to the current Node.js process. The [Public Variables](#public-variables) will be exposed in the client code.
```

```ts
import { loadEnv, mergeRsbuildConfig } from '@rsbuild/core';

> // Here publicVars is the PUBLIC_ prefixed variable read by default.

Reviewer: ```suggestion
// By default, `publicVars` are variables prefixed with `PUBLIC_`
```
